core/src/main/java/org/ehcache/EhcacheManager.java

```diff
@@ -496,13 +498,25 @@ public void create() {
 
         @Override
         public void destroy() {
+          startPersistenceService();
```

Comment:
As starting locks the persistence location and failing to do so indicates it is used by someone else, it must be part of "going to maintenance" because you can't go into maintenance unless that succeeds.
So starting and stopping must happen as part of going and leaving maintenance.